bee-network/src/backstage/peer.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    message::MessageType,
+    peer::{Error, PeerInfo, PeerReader, PeerWriter},
+};
+
+use backstage::core::{AbortableUnboundedChannel, Actor, ActorResult, NullChannel, Rt, StreamExt, SupHandle};
+
+use std::sync::Arc;
+
+/// A peer reader worker.
+pub struct PeerReaderWorker {
+    reader: PeerReader,
+    info: Arc<PeerInfo>,
+}
+
+impl PeerReaderWorker {
+    pub(crate) fn new(reader: PeerReader, info: Arc<PeerInfo>) -> Self {
+        Self { reader, info }
+    }
+}
+
+#[async_trait::async_trait]
+impl<S: SupHandle<Self>> Actor<S> for PeerReaderWorker {
+    type Data = ();
+
```

Comment:
Do you mind not putting new lines between associated types ? This is inconsistent with the rest of the codebase.

Reply:
I think Christian doesn't. I'll remove the empty line.